packages/maplestory-open-api-js/src/ranking.ts

```diff
@@ -0,0 +1,205 @@
+import { get } from './fetcher';
+import {
+  AchievementRankingQueryParameter,
+  DojangRankingQueryParameter,
+  GuildRankingQueryParameter,
+  OverallRankingQueryParameter,
+  TheSeedRankingQueryParameter,
+  UnionRankingQueryParameter,
+} from './types/params';
+import {
+  AchievementRanking,
+  DojangRanking,
+  GuildRanking,
+  OverallRanking,
+  TheSeedRanking,
+  UnionRanking,
+} from './types/ranking';
+import { RANKING_URLS } from './urls';
+import { getPreviousFormattedDate } from './utils';
+
+/**
+ * @description 종합 랭킹 정보를 조회
+ * @param ocid 캐릭터 식별자
+ * @param character_class 직업 및 전직
+ * @param world_name 월드 명
+ * @param world_type 월드 타입 (0:일반, 1:리부트) (기본 값은 0이며, world_name 입력 시 미 반영)
+ * @param page 페이지 번호
+ * @param date 조회 기준일 (KST, EXAMPLE: 2023-12-21)
+ */
+export function getOverallRanking({
+  ocid,
+  character_class,
+  world_name,
+  world_type = 0,
+  page,
+  date = getPreviousFormattedDate(),
+}: OverallRankingQueryParameter = {}) {
+  const searchParams: {
+    ocid?: string;
+    character_class?: string;
+    world_name?: string;
+    world_type?: number;
+    page?: number;
+    date?: string;
+  } = { date, world_type };
```

Comment:
이렇게 사용해도 괜찮지 않을까요? 새로운 객체를 굳이 만들지 않아도 괜찮을 것 같긴하네요~!

```suggestion
  const searchParams: OverallRankingQueryParameter = { date, world_type };
```

Reply:
저도 처음에 이 방식으로 해 보려 했는데, 다음과 같은 오류가 발생했습니다.
![image](https://github.com/alstn2468/maplestory-open-api-js/assets/102154824/1c0a7cf0-ae6c-423f-ba29-27ff2ce1a0f7)
그래서 새 객체를 만들고 타입을 따로 지정해 주는 방식으로 했었는데, 혹시 더 나은 방법이 있을까요?

Reply:
이렇게 작성해도 괜찮을 것 같아요!

```ts
export function getOverallRanking({
  ocid,
  character_class,
  world_name,
  world_type = 0,
  page,
  date = getPreviousFormattedDate(),
}: OverallRankingQueryParameter) {
  return get<OverallRanking>(RANKING_URLS.OVERALL, {
    searchParams: {
      ocid,
      character_class,
      world_name,
      page,
      world_type,
      date,
    },
  });
}
```

Reply:
![image](https://github.com/alstn2468/maplestory-open-api-js/assets/30997311/df137e84-cd68-4d69-b09c-52919c5a2aa5)

음 제 로컬에서는 따로 문제는 없는 것 같긴합니다..! 또 필수값이라 파라미터 기본값을 `{}` 로 지정해주지 않아도 괜찮을 것 같아요!

Reply:
기본값을 `{}` 로 지정한 이유는 해당 함수를 아무 값도 전달하지 않고 호출할 때 `{}` 를 넣어줘야 작동하더라구요..! 
```ts
const data = await getOverallRanking() // X
const data = await getOverallRanking({ }) // O 
```
이 부분을 고려해서 `{}` 를 기본값으로 주었습니다! 이 부분은 어떻게 생각하시는지 궁금합니다..!

Reply:
아하 이해했습니다! 필수 파라미터가 없는 경우 그럴 수 있겠군요! 좋은 것 같습니다!

Reply:
> ![image](https://private-user-images.githubusercontent.com/30997311/294657732-df137e84-cd68-4d69-b09c-52919c5a2aa5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1MzE5MDcsIm5iZiI6MTcwNDUzMTYwNywicGF0aCI6Ii8zMDk5NzMxMS8yOTQ2NTc3MzItZGYxMzdlODQtY2Q2OC00ZDY5LWIwOWMtNTI5MTljNWEyYWE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA2VDA5MDAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyZDJhZmJjOWRlNTc1ZThjYzdmOGRiN2U0NmVjZWM1ZDVlNDMyMTNiNTQ5ZDUzYTQwNDgyYTYwODYzYjViMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2I7EDdVolrE3d0iEP9Ncy9m9gBmNPp5InP8_U32H9vI)
> 
> 음 제 로컬에서는 따로 문제는 없는 것 같긴합니다..! 또 필수값이라 파라미터 기본값을 `{}` 로 지정해주지 않아도 괜찮을 것 같아요!

답변감사합니다!! 그러면 이 방식으로 수정 해 보겠습니다.!